vcsclient/github.go

```diff
@@ -336,27 +336,53 @@ func (client *GitHubClient) GetPullRequestByID(ctx context.Context, owner, repos
 		}
 	}
 
-	sourceBranch, err1 := extractBranchFromLabel(vcsutils.DefaultIfNotNil(pullRequest.Head.Label))
-	targetBranch, err2 := extractBranchFromLabel(vcsutils.DefaultIfNotNil(pullRequest.Base.Label))
-	err = errors.Join(err1, err2)
-	if err != nil {
-		return PullRequestInfo{}, err
+	return mapGitHubPullRequestToPullRequestInfo(pullRequest, false)
+}
+
+func mapGitHubPullRequestToPullRequestInfo(ghPullRequest *github.PullRequest, withBody bool) (PullRequestInfo, error) {
+	var sourceBranch, targetBranch string
+	var err1, err2 error
+	if ghPullRequest != nil && ghPullRequest.Head != nil && ghPullRequest.Base != nil {
+		sourceBranch, err1 = extractBranchFromLabel(vcsutils.DefaultIfNotNil(ghPullRequest.Head.Label))
+		targetBranch, err2 = extractBranchFromLabel(vcsutils.DefaultIfNotNil(ghPullRequest.Base.Label))
+		err := errors.Join(err1, err2)
+		if err != nil {
+			return PullRequestInfo{}, err
+		}
+	}
+
+	var sourceRepoName, sourceRepoOwner string
+	if ghPullRequest.Head.Repo != nil && ghPullRequest.Head.Repo.Owner != nil {
```

Comment:
why?
what happens if its not true?

Reply:
Then we'll send empty values, do you think we need to throw an error in these cases? @sverdlov93